azure-pipelines.yml

```diff
@@ -86,6 +97,14 @@ jobs:
   pool:
     vmImage: 'windows-latest'
   steps:
+  - bash: |
+      curl -fsSL --output "$TEMP/go.zip" "$GO_ARCHIVE_URL"
+      echo "$GO_ARCHIVE_CHECKSUM $TEMP/go.zip" | sha256sum --check
+      unzip "$TEMP/go.zip" -d "/usr/bin/go.1.16"
+      ln -s /usr/bin/go.1.16/bin/go /usr/bin/go
```

Comment:
Seems like we're running into git bash symlinking issues

Could be worth it to just do what we did here: https://github.com/envoyproxy/envoy-build-tools/blob/9f058c4f2b24de42da5e965f878d4ded865a1be4/azure-pipelines.yml#L69 (add the relevant directory to the path) or use `cmd` to make the symlink, or try messing with the git bash configs and setting `winsymlinks:nativestrict`

See:
- https://www.joshkel.com/2018/01/18/symlinks-in-windows/
- https://www.cygwin.com/cygwin-ug-net/using.html#pathnames-symlinks